src/big_core_input.cpp

```diff
@@ -85,6 +89,9 @@ template<typename T>
 const T& BigCoreInput::operator()(uint64_t imageNum, uint64_t row, uint64_t column, uint64_t plane, uint64_t tileNum) const
 {
 	uint64_t dataTypeIndex = this->isUniformDataType() ? 0 : imageNum;
+	// TODO: this must be adjusted according to data order. Would only work for image being outermost entity
+	// part this->data[this->outermostEntitiesOffsets[imageNum]] is alright, as it points directly to the first
+	// element of the image. The rest is wrong.
 	return static_cast<const T>(this->data[this->outermostEntitiesOffsets[imageNum] + row * column * plane * tileNum * this->getImageType(this->dataType[dataTypeIndex])]);
```

Comment:
Tady je trochu chyba, že místo
this->data[this->outermostEntitiesOffsets[imageNum]
musí být
this->data[this->outermostEntitiesOffsets[firstDimNum],
kde firstDimNum je jeden z těch pěti parametrů podle dataOrder.